feat: introduced initializer predicate

OlegErshov opened the pull request:
On-behalf-of: SAP aleh.yarshou@sap.com

This pr is related to #175.

Changes:
1. Introduced a controller which reconciles logical cluster resource and apply the same logic as initializer does
2. Introduced predicate to filter logical clusters and reconcile only logical clusters which have been initialized by security-operator

OlegErshov commented on internal/controller/accountlogicalcluster_initializer_controller.go:
```diff
@@ -0,0 +1,51 @@
+	log *logger.Logger
+
+	mclifecycle *multicluster.LifecycleManager
+}
```
addressed

OlegErshov commented on internal/predicates/initializer.go:
```diff
@@ -0,0 +1,26 @@
+		lc, ok := object.(*kcpcorev1alpha1.LogicalCluster)
+		if !ok {
+			panic(fmt.Errorf("received non-LogicalCluster resource in LogicalClusterIsAccountTypeOrg predicate"))
+		}
```
addressed

simontesar reviewed (commented):
I am late to the discussion here and am fine with merging this for now in case it is a priority.

What bothers me that we now have two reconcilers instead of one but that could actually be simplified to one by passing predicates and lifecycle options to the constructor, could it not?

Also, `Processor`, `Finalizer` and `Terminator` interface implementations now referring to a `reconcile` function in my opinion misses the point of the `subroutines` package and seems like working around a bad architecture.

I'm wondering if this is something that should be integrated into https://github.com/platform-mesh/subroutines in some way, e.g. be able to use `WithReconcilingTerminator("foo")` on a single controller that achieves what we're doing here manually with multiple reconcilers.

OlegErshov commented:
Thanks for review! I've updated the way controllers are configured and removed extra files after it. 

Currently `reconcile`  function is used just to hold common code between `Initializer` and `Processor` interfaces, `Terminator` one is untouched.  I think if we want to get rid of `reconcile`  function  we need to change how lifecycle works with `Processor` and `Initializer` interfaces. I reckon it's better to do in a follow up pr
